client/my-sites/site-settings/cloudflare-analytics/index.js

```diff
@@ -31,62 +31,64 @@ import './style.scss';
 
 const validateTrackingId = ( code ) => ! code || code.match( /^(UA-\d+-\d+)|(G-[A-Z0-9]+)$/i ); // @TODO: how are cloudflare tracking IDs formatted?
 
-export class CloudflareAnalyticsSettings extends Component {
-	state = {
-		isCodeValid: true,
-		loggedCloudflareAnalyticsModified: false,
-	};
-
-	handleFieldChange = ( key, value ) => {
-		const { fields, updateFields } = this.props;
+export function CloudflareAnalyticsSettings( {
+	fields,
+	updateFields,
+	isRequestingSettings,
+	isSavingSettings,
+	enableForm,
+	eventTracker,
+	handleSubmitForm,
+	path,
+	translate,
+	trackTracksEvent,
+	uniqueEventTracker,
+	site,
+} ) {
+	const [ isCodeValid, setIsCodeValid ] = useState( true );
+	const [ loggedCloudflareAnalyticsModified, setLoggedCloudflareAnalyticsModified ] = useState(
+		false
+	);
+	const isSubmitButtonDisabled =
+		isRequestingSettings || isSavingSettings || ! isCodeValid || ! enableForm;
+
+	const handleFieldChange = ( key, value ) => {
```

Comment:
The "modern" equivalent of this to prevent unnecessary re-renders is to use the `useCallback` hook, if you want